test/integration/http_integration.cc

```diff
@@ -867,6 +867,32 @@ void HttpIntegrationTest::testEnvoyProxying100Continue(bool continue_before_upst
   EXPECT_EQ("200", response->headers().Status()->value().getStringView());
 }
 
+void HttpIntegrationTest::testResponseFramedByConnectionCloseWithReadLimits(
+    ConnectionCreationFunction* create_connection) {
+  // Set a small buffer limit on the downstream in order to trigger a call to trigger readDisable on
+  // the upstream when proxying the response. Upstream limit needs to be larger so that
+  // RawBufferSocket::doRead reads the response body and detects the upstream close in the same call
+  // stack.
+  config_helper_.setBufferLimits(100000, 1);
+  initialize();
+
+  codec_client_ = makeHttpConnection(
+      create_connection ? ((*create_connection)()) : makeClientConnection((lookupPort("http"))));
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+
+  auto response = codec_client_->makeHeaderOnlyRequest(default_request_headers_);
+  waitForNextUpstreamRequest();
+  // Disable chunking to trigger framing by connection close.
+  upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}, {":no-chunks", "1"}},
```

Comment:
This is just run for HTTP/1.1 right?
ostensibly the no chunk header is never supposed to be sent on the wire - I think if you don't send a content length and send HTTP/1.0 (see existing example tests) you can force frame by close that way. Mind trying it out?

Reply:
I'm having some trouble finding examples under test/integration that encode an HTTP/1.0 response.  Do you have some links handy?

As far as I can tell ":no-chunks" is the way to tell StreamEncoderImpl::encodeHeaders in source/common/http/http1/codec_impl.cc to not chunk a response without a content-length.

Reply:
grep around for config_helper_.addConfigModifier(&setAllowHttp10WithDefaultHost); ?

They all use sendRawHttpAndWaitForResponse but given this is a header only request and you're waiting for connection close I _think_ that should be fine.

Reply:
I'm fairly sure that the :no-chunk does not end up on the wire.  One thing that I wanted to verify in some way is that the Envoy does see an upstream response framed by connection close.  I'm not sure how to accomplish that, since I think envoy will add chunk encoding to the response it sends downstream.

Let's talk briefly offline.